andela/ah-django-unchained
# #164191790 Return average rating in get single articles and get all articles

**What does this PR do?**

To enable fetching articles' average ratings in the frontend, this PR
- adds average_rating field in article model
- refactors articles view to return average rating
- refactors article ratings tests

**How should this be manually tested?**

* Clone the repo locally
* cd `ah-django-unchained`
* git checkkout `bg-return-average-rating-164191790`
* get started with the app as from the README section
* to add an article ensure you obtain a token from logging in or signing up if not signed up
* Using the `POST api/articles/` endpoint, have the following JSON object. Note that image uploads are optional.
```
{
	"title": "iphone",
	"description": "here wbvhjcome",
	"body": "You have to believe",
	"tagList": ["dragons", "training"]
}
```
You should get the following response
```
{
    "title": "iphone",
    "slug": "iphone",
    "description": "here wbvhjcome",
    "body": "You have to believe",
    "is_published": false,
    "created": "2019-02-24T12:05:30.004612Z",
    "modified": "2019-02-24T12:05:30.004648Z",
    "images": null,
    "author": "jameswafula",
    "tagList": [
        "dragons",
        "training"
    ],
    "user_id_likes": [],
    "user_id_dislikes": [],
    "likes_count": 0,
    "dislikes_count": 0,
    "favorite": [],
    "average_rating": 0
}
```
Notice the `average_rating field`.

To view a single article, make a `GET` request on url `http://127.0.0.1:8000/api/articles/<slug>`. Below is a sample response.
```
{
    "title": "very lovely",
    "description": "here wbvhjcome",
    "body": "You have to believe",
    "modified": "2019-02-24T11:40:36.919633Z",
    "images": "http://res.cloudinary.com/authors/image/upload/No%20image",
    "author": "andrew",
    "is_published": true,
    "slug": "very-lovely-5026",
    "tagList": [
        "dragons",
        "training"
    ],
    "favorite": false,
    "average_rating": 4
}
```

**Relevant Pivotal Tracker Story**

[#164191790](https://www.pivotaltracker.com/story/show/164191790)
